src/mono/wasm/Wasm.Build.Tests/Common/ToolCommand.cs

```diff
@@ -133,7 +133,7 @@ private async Task<CommandResult> ExecuteAsyncInternal(string executable, string
             CurrentProcess.BeginOutputReadLine();
             CurrentProcess.BeginErrorReadLine();
             await completionTask;
-
+            CurrentProcess.Kill();
```

Comment:
I imagine that some child processes would be able to shutdown gracefully. 
Could we `await` graceful shutdown with timeout, say `1000`ms and only kill if that didn't resolve in time ?

Reply:
Yeah, I would like to understand both why the process isn't exiting and why that is causing the next run to hang.  Is there a problem in the msbuild or compiler services?

Reply:
`completionTask` should really be completing only when the `Exited` event has been raised. You could add a `await CurrentProcess.WaitForExitAsync(cancellationToken)` after that to be sure we complete that side.

Also, I would suggesting dumping the process status and debugging this a bit. When I was looking at it, there was some odd behavior.

Reply:
CI does not fail anymore, even with parallel tests runs. We should unblock the tests.